numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -958,6 +958,20 @@ PyArray_MatrixProduct(PyObject *op1, PyObject *op2)
     return PyArray_MatrixProduct2(op1, op2, NULL);
 }
 
+static int
+check_fperr(const char *func_name)
+{
+    int errmask = 0;
+    if (_get_bufsize_errmask(NULL, &errmask) < 0) {
+        return -1;
+    }
+    if (_check_ufunc_fperr(errmask, func_name) < 0) {
+        return -1;
+    }
+
+    return 0;
+}
```

Comment:
Please use `PyUFunc_GiveFloatingpointErrors`, you can find the rest of the pattern elsewhere (e.g. in scalar math).